### PR TITLE
SQL: Provide null-safe scripts for Not and Neg

### LINKED ISSUE #34848: SQL: Implement NOT behaviour
Currently NOT cannot be used inside 'scripting contexts' (such as `HAVING`) or against native negateable operators.
Clearly this needs to be fixed.

Relates #34558 #34680

### PR DESCRIPTION
**Pull request by costin**
Introduce null-safe Painless scripts for Not and Neg
Simplify script generation for Unary functions

Close #34848

### CODE REVIEW

File: x-pack/qa/sql/src/main/resources/agg.sql-spec [hunk 1]
```diff
@@ -456,6 +456,13 @@ selectHireDateGroupByHireDate
 SELECT hire_date HD, COUNT(*) c FROM test_emp GROUP BY hire_date ORDER BY hire_date DESC;
 selectSalaryGroupBySalary
 SELECT salary, COUNT(*) c FROM test_emp GROUP BY salary ORDER BY salary DESC;
+selectSalaryGroupBySalaryHavingIsNotNull
+SELECT salary, COUNT(*) c FROM test_emp GROUP BY salary HAVING MAX(languages) IS NOT NULL ORDER BY salary DESC;
```
Comment (astefan) [inline x-pack/qa/sql/src/main/resources/agg.sql-spec]: This is a good test atm, but not really testing the behavior as MAX() doesn't _seem_ to return `null` in any scenario https://github.com/elastic/elasticsearch/issues/34896

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/qa/sql/src/main/resources/agg.sql-spec (hunk 1 above)
Comment (costin) [reply x-pack/qa/sql/src/main/resources/agg.sql-spec]: Good catch - due the use of primitives, a non-null value is returned all the time.

**Review by matriv — APPROVED** [review]
LGTM! left a question

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/logical/NotProcessor.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        return apply(input);
+    }
+
+    public static Boolean apply(Object input) {
+        if (input == null) {
+            return null;
+        }
+        
+        if (!(input instanceof Boolean)) {
+            throw new SqlIllegalArgumentException("A boolean is required; received {}", input);
```
[the remark this answers is not in the document]
Comment (costin) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/logical/NotProcessor.java]: It is yes - we might not need them  long term.

**Pull request merged**